DABI.js

```diff
@@ -0,0 +1,321 @@
+{
+    "translatorID": "5cf8bb21-e350-444f-b9b4-f46d9fab7827",
+    "label": "DABI",
+    "creator": "Jens Mittelbach",
+    "target": "^https?://dabi\\.ib\\.hu-berlin\\.de/",
+    "minVersion": "1.0",
+    "maxVersion": "",
+    "priority": 100,
+    "inRepository": true,
+    "translatorType": 4,
+    "browserSupport": "gcsibv",
+    "lastUpdated": "2016-01-08 11:43:21"
+}
+
+function detectWeb(doc, url) {
+    if (doc.title.trim().indexOf("DABI. Datensatz Vollanzeige") == 0) {
+        return "journalArticle";
+    } else if (doc.title.trim().indexOf("DABI: Rechercheergebnis") == 0) {
+        var keinTreffer = doc.getElementsByTagName("br")[7].nextSibling.data.indexOf("Keine Treffer für die Suche nach:")
+        if (keinTreffer == -1) {
+            return "multiple";
+        }
+    }
+}
+
+function doWeb(doc, url) {
+    var ids = [];
+
+    //If Statement checks if page is a Search Result, then saves requested Items
+    if (detectWeb(doc, url) == "multiple") {
+        Z.selectItems(getSearchResults(doc), function(items) {
+            if (!items) return true;
+            for (var i in items) {
+                ids.push(i);
+            }
+        });
+    } else if (detectWeb(doc, url) == "journalArticle") {
+        //saves single page items
+        ids = [url];
+    }
+    ZU.processDocuments(ids, scrape);
+}
+
+
+function getSearchResults(doc) {
+    var trs = doc.getElementsByTagName("tr"),
+        tds = null,
+        items = {};
+
+    for (var i = 1; i < trs.length; i++) {
+        tds = trs[i].getElementsByTagName("td");
+        for (var n = 0; n < tds.length; n++) {
+            var url = doc.location.origin + "/cgi-bin/dabi/" + tds[0].firstChild.getAttribute("href"),
+                author = tds[1].innerHTML,
+                title = tds[2].innerHTML.replace(/<br>/g, '. ');
+
+            if (!author == '') {
+                var item = author.replace(/; <br>.*/, ' et al.') + ": " + title;
+            } else {
+                var item = title;
+            }
+            if (!item || !url) continue;
+
+            items[url] = item;
+        }
+    }
+    return items;
+}
+
+function scrape(doc, url) {
+    var newItem = new Zotero.Item('journalArticle');
+    var trs = doc.getElementsByTagName("tr"),
+        headers,
+        contents,
+        items = {};
+
+    //For Loop to populate "items" Object and save tags to an Array.
+    for (var i = 0; i < trs.length; i++) {
+        headers = trs[i].getElementsByTagName("th")[0].textContent;
+        contents = trs[i].getElementsByTagName("td")[0].innerHTML;
+
+        items[headers.replace(/\s+/g, '')] = contents.trim();
+    }
+
+    //set url to fulltext resource, if present; else to database item
+    if (items["URL"] == '') {
+        newItem.url = url;
```

Comment:
URL should always point to the full-text resource. It should not point to a DB entry where the full text is not available.

Reply:
I don't think that is the case. The DB does not host PDFs. Your example seems to have changed the ID (now http://dabi.ib.hu-berlin.de/cgi-bin/dabi/vollanzeige.pl?artikel_id=21283&modus=html), the full text resource is (and was?) elsewhere. The fact that ID changes are possible lets me argue that referencing the actual PDF when availble is most reasonable.

I'm going to change the handling for metadata-only entries. These will not include URLs any longer.